### ISSUE TITLE
Found fix?

### ISSUE THREAD

**Issue by GunterO** [JonBAL/STM32-ChibiOS-uIP-ENC28J60#1]
Hello,

I read about your problem with this project on the ChibiOS forum.
I wonder if you eventually came up with a fix? I'm trying to get a STM32 + ENC28J60 working as well.

Thanks!

**Comment by JonBAL** [comment]
@GunterO 
Recent fixes and changes like reduced the number of deadlocks.
I studied that gives DBG at these lockups, and it shows on the functions to work with SPI. I think you need to redo work with SPI, maybe add a queue or mutex. But I'm not good at ChibiOS and it will require a lot of time.

I took a pause on this project, but now come back to it.
You want to use the same bundle: STM32 + uIP + ENC28J60 ?? What you have success with this?

**Comment by GunterO** [comment]
Yes, I have a board with a STM32 + ENC28J60 I want to use for a project, and hoped to use ChibiOS + uIP + µGUI for it. And all this in Keil µVision :-)
But I can't find much info/help about it, except yours. I did some research and I'm affraid I'll have to change the board to a STM32F207 of F407 and use the onboard Ethernet + PHY to have a reliable solution. I have some experience with the ENC28J60 in the Arduino world, and the libraries over there took some time to get stable as well. I think that you're better off with a Wiznet W5100/W5500 then an ENC28J60 because it has a build-in TCP stack.
But, preferable, I wish there was a reliable STM32 + ENC28J60 + ChibiOS + uIP solution out there :-)
